File: src/main/java/toast/pine/Classes/Quests/Quest.java
package toast.pine.Classes.Quests;

import toast.pine.SocialSystem.PlayerSocial;

public class Quest {

    private PlayerSocial questOwner;
    private String name;
    private String description;
    private QuestObjective objective;
    private boolean completed;

    public Quest(PlayerSocial owner, String name, String description, QuestObjective objective) {
        this.questOwner = owner;
        this.name = name;
        this.description = description;
        this.objective = objective;
        this.completed = false;
    }
    
    public void updateProgress() {
        if(objective.isComplete()) {
            completed = true;
        } else objective.updateProgress();
    }


}

File: src/main/java/pine/toast/Classes/Quests/Quest.kt
package toast.pine.Classes.Quests

import toast.pine.socialSystem.PlayerSocial

class Quest(owner: PlayerSocial, name: String, description: String, objective: QuestObjective) {
    private val questOwner: PlayerSocial
    private val name: String
    private val description: String
    private val objective: QuestObjective
    private var completed: Boolean

    init {
        questOwner = owner
        this.name = name
        this.description = description
        this.objective = objective
        completed = false
    }

    fun updateProgress() {
        if (objective.isComplete) {
            completed = true
        } else objective.updateProgress()
    }
}
